paimon-core/src/main/java/org/apache/paimon/io/KeyValueFileReaderFactory.java

```diff
@@ -144,7 +144,8 @@ private FileRecordReader<KeyValue> createRecordReader(
                                         fileIO, filePath, fileSize, orcPoolSize),
                         bulkFormatMapping.getIndexMapping(),
                         bulkFormatMapping.getCastMapping(),
-                        PartitionUtils.create(bulkFormatMapping.getPartitionPair(), partition));
+                        PartitionUtils.create(bulkFormatMapping.getPartitionPair(), partition),
+                        bulkFormatMapping.getTrimmedKeyMapping());
```

Comment:
Can we just merge this trimmedKeyMapping into indexMapping?

Reply:
Sure, we can